src/Simplex/Chat/Store/Messages.hs

```diff
@@ -967,11 +1104,88 @@ getGroupChatBefore_ db user@User {userId} g@GroupInfo {groupId} beforeChatItemId
           |]
           (userId, groupId, search, beforeChatItemTs, beforeChatItemTs, beforeChatItemId, count)
 
-toChatItemRef :: (ChatItemId, Maybe Int64, Maybe Int64) -> Either StoreError (ChatRef, ChatItemId)
+getLocalChat :: DB.Connection -> User -> Int64 -> ChatPagination -> Maybe String -> ExceptT StoreError IO (Chat 'CTLocal)
+getLocalChat db user folderId pagination search_ = do
+  let search = fromMaybe "" search_
+  nf <- getNoteFolder db user folderId
+  case pagination of
+    CPLast count -> getLocalChatLast_ db user nf count search
+    CPAfter afterId count -> getLocalChatAfter_ db user nf afterId count search
+    CPBefore beforeId count -> getLocalChatBefore_ db user nf beforeId count search
+
+getLocalChatLast_ :: DB.Connection -> User -> NoteFolder -> Int -> String -> ExceptT StoreError IO (Chat 'CTLocal)
+getLocalChatLast_ db user@User {userId} nf@NoteFolder {noteFolderId} count search = do
+  let stats = ChatStats {unreadCount = 0, minUnreadItemId = 0, unreadChat = False}
+  chatItemIds <- liftIO getLocalChatItemIdsLast_
+  chatItems <- mapM (getLocalChatItem db user noteFolderId) chatItemIds
+  pure $ Chat (LocalChat nf) (reverse chatItems) stats
+  where
+    getLocalChatItemIdsLast_ :: IO [ChatItemId]
+    getLocalChatItemIdsLast_ =
+      map fromOnly
+        <$> DB.query
+          db
+          [sql|
+            SELECT chat_item_id
+            FROM chat_items
+            WHERE user_id = ? AND note_folder_id = ? AND item_text LIKE '%' || ? || '%'
+            ORDER BY item_ts DESC, chat_item_id DESC
```

Comment:
better to repeat logic from direct chats here and in other chat pagination functions (especially it affects before and after queries by removing some extra queries and filters)
```suggestion
            ORDER BY created_at DESC, chat_item_id DESC
```

Reply:
groups have more difficult pagination logic because messages arrive from different connections in arbitrary order, and since we don't maintain causality graph there yet, there's currently no better ordering criteria than item_ts (which is also not perfect but better than created_at)